plugins/woocommerce-docs/src/Manifest/PostCreator.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace WooCommerceDocs\Manifest;
+
+use WooCommerceDocs\Data\DocsStore;
+use WooCommerceDocs\Blocks\BlockConverter;
+
+/**
+ * Class PostCreator
+ *
+ * Create a post from a manifest entry.
+ *
+ * @package WooCommerceDocs\Manifest
+ */
+class PostCreator {
+	/**
+	 * Get the Markdown converter.
+	 */
+	private static function get_converter() {
+		static $converter = null;
+		if ( null === $converter ) {
+			$converter = new BlockConverter();
+		}
+		return $converter;
+	}
+
+	/**
+	 * Create a post from a manifest entry and file contents.
+	 *
+	 * @param mixed  $manifest_post The manifest representation of the post.
+	 * @param string $post_content The post content as a string.
+	 * @param int    $logger_action_id The logger action ID.
+	 * @return int The post ID.
+	 */
+	public static function create_post_from_manifest_entry( $manifest_post, $post_content, $logger_action_id ) {
+		$existing_post = DocsStore::get_post( $manifest_post['id'] );
+
+		// Strip frontmatter.
+		$content = preg_replace( '/^---[\s\S]*?---/', '', $post_content );
```

Comment:
I extracted this into the BlockConverter so it can handle the frontmatter stripping itself. Covered with tests too. https://github.com/woocommerce/woocommerce/pull/39241/commits/793d00291aef19b54ac3c17bb689f68de66cac39